aleph0/index.ts

```diff
@@ -1 +1,87 @@
-console.log(process.argv)
+import { Command } from "commander";
+import { getProjectStructure, loadSnippets } from "./utils";
+import { createChangesArray, findRelevantSnippets } from "./prompts";
+import { ai } from "./openai";
+
+async function generate(text: string) {
+  console.log("Creating:", text);
+
+  // 1. Find the relevant snippet
+  const snippets = await loadSnippets();
+  const relevantSnippet = await ai(findRelevantSnippets(text, snippets));
+  if (!relevantSnippet)
+    throw new Error(`AI returned a bad snippet: ${relevantSnippet}`);
+
+  console.log("Found snippet:", relevantSnippet);
+
+  // 2. Find the relevant files we are dealing with
+  // For each snippet file, find the corresponding file to be created/modified
+  // const changes = [{snippet: 'path', sourceFile: 'path'}]
+  const projectSturcutre = await getProjectStructure("../examples/next");
+  const changesRaw = await ai(
+    await createChangesArray(relevantSnippet, projectSturcutre),
+    "Find what files are relevant for theseS snippets in this project.",
+    "gpt-4"
+  );
+  if (!changesRaw)
+    throw new Error(`AI returned a bad changes array: ${relevantSnippet}`);
+
+  const changes = JSON.parse(changesRaw);
+  console.log(changes);
+
+  // 3. For each file in the changes array, ask GPT 4 for the new file and create/modify it.
+  // for(const change of changes) {
+  //   if(change.create)
+  //   fs.mkdirSync(<the path to create>, { recursive: true });
+  // }
+}
+
+const program = new Command();
+program
+  .command("gen <text>")
+  .description("Generate nextjs snippet.")
+  .action((text) => {
+    generate(text);
+  });
+program.parse(process.argv);
+
+// example to create an "article" component with an api route
```

Comment:
feel free to delete this code when you want btw

Reply:
Not yet.